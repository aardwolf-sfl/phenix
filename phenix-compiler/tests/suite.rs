use std::path::Path;

use phenix_codegen::Project;
use phenix_compiler::Compiler;

fn compile<P: AsRef<Path>>(path: P) -> Project {
    Compiler::new().compile(path.as_ref().try_into().unwrap())
}

#[test]
fn basic() {
    let project = compile("tests/schemas/basic/index.phenix");
    insta::assert_yaml_snapshot!(project);
}

#[test]
fn imports() {
    let project = compile("tests/schemas/imports/index.phenix");
    insta::assert_yaml_snapshot!(project);
}
