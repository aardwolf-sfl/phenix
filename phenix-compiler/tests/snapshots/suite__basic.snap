---
source: phenix-compiler/tests/suite.rs
expression: project
---
modules:
  - id: 0
    path: []
    types:
      - type: Struct
        id: 0
        name: Struct
        fields:
          - name: number
            ty:
              id:
                builtin: uint
              generics: []
            attrs: []
          - name: booleans
            ty:
              id:
                builtin: vector
              generics:
                - id:
                    builtin: bool
                  generics: []
            attrs: []
        attrs: []
      - type: Enum
        id: 1
        name: Enum
        variants:
          - name: Bare
            fields: []
            attrs: []
          - name: Rich
            fields:
              - name: value
                ty:
                  id:
                    builtin: string
                  generics: []
                attrs: []
            attrs: []
        attrs: []
      - type: Flags
        id: 2
        name: Flags
        flags:
          - FOO
          - BAR
          - BAZ
        attrs: []

